frontend/src/pages/admin/users/ManageUsers.tsx

```diff
@@ -12,55 +11,101 @@ import './ManageUsers.scss';
 import { UsersList, TableStateChangeFn } from './components/UsersList';
 import { IUser } from 'interfaces';
 import { MUIDataTableState } from 'mui-datatables';
-import { IUserRecord, AccountActive } from './interfaces/IUserRecord';
+import { IUserRecord, AccountActive, accountActiveToBool } from './interfaces/IUserRecord';
+import { IUsersState, MAX_USERS_PER_PAGE } from 'reducers/usersReducer';
+import { setUsersFilter, setUsersPageSize, setUsersSort, IUsersSort } from 'actions/adminActions';
+import { sortBy } from 'lodash';
 
 const ManageUsers = () => {
   const dispatch = useDispatch();
-  const MAX_USERS_PER_PAGE = 10;
   const pagedUsers = useSelector<RootState, IPagedItems<IUser>>(state => {
-    return (state.users as IStoreUsersAction).pagedUsers;
+    return (state.users as IUsersState).pagedUsers;
   });
+
+  const pageSize = useSelector<RootState, number>(state => {
+    return (state.users as IUsersState).rowsPerPage;
+  });
+
+  const sort = useSelector<RootState, IUsersSort>(state => {
+    return (state.users as IUsersState).sort;
+  });
+
   const users = useSelector<RootState, IGenericNetworkAction>(
     state => (state.network as any)[actionTypes.GET_USERS] as IGenericNetworkAction,
   );
   useEffect(() => {
     dispatch(getUsersAction(toApiPaginateParams(0, MAX_USERS_PER_PAGE)));
-
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   const tableChange: TableStateChangeFn = (type: string, state: MUIDataTableState) => {
     switch (type) {
-      case 'pagination':
-        break;
-      case 'filter':
-        break;
       case 'sort':
+        const sortColumn = state.columns.find(x => x.sortDirection !== 'none');
+        if (sortColumn) {
+          dispatch(
+            setUsersSort({
+              sortBy: sortColumn.name as any,
+              direction: sortColumn.sortDirection as any,
+            }),
+          );
+        }
+        break;
+      case 'resetFilters':
+      case 'changePage':
+      case 'filterChange':
+      case 'changeRowsPerPage':
+        dispatch(setUsersFilter(state.filterList));
+        dispatch(setUsersPageSize(state.rowsPerPage));
+
+        dispatch(
+          getUsersPaginationAction({
+            page: state.page + 1,
+            quantity: state.rowsPerPage,
+            username: state.filterList[1][0] || '',
```

Comment:
I'm not a big fan of using a two dimensional array to get values like this, but it may not matter if we are swapping table implementations anyhow.

Reply:
Yeah, this how the grid presents the filter values